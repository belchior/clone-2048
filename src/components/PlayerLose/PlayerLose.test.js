import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { PlayerLose } from './index';

const initialProps = {
  bestScore: 0,
  button: { text: '', action: () => {}},
  hardMode: false,
  score: 0,
  wall: [],
};

const renderer = new ShallowRenderer();

it('PlayerLose should render without crashing', () => {
  const tree = renderer.render(<PlayerLose {...initialProps} />);
  expect(tree).toMatchSnapshot();
});

it('PlayerLoseAction button when clicked should call buttonAction callback', () => {
  const buttonAction = jest.fn();
  const props = {
    ...initialProps,
    button: { text: 'Action', action: buttonAction },
  };
  const tree = renderer.render(<PlayerLose {...props} />);
  const playerLoseAction = tree.props.children[1].props.children[2].props;

  playerLoseAction.onClick();
  expect(buttonAction.mock.calls).toHaveLength(1);
});
