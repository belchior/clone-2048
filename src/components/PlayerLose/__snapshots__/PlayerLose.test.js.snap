// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerLose should render without crashing 1`] = `
<section
  className="PlayerLose"
>
  <Wall
    hardMode={false}
    wall={Array []}
  />
  <div
    className="PlayerLose-content"
  >
    <h1
      className="PlayerLose-title"
    >
      You Lose
    </h1>
    <Score
      best={0}
      current={0}
    />
    <button
      className="PlayerLoseAction"
      onClick={[Function]}
      type="button"
    />
  </div>
</section>
`;
